pennylane/devices/qubit/simulate.py

```diff
@@ -125,16 +137,25 @@ def get_final_state(circuit, debugger=None, interface=None):
     # initial state is batched only if the state preparation (if it exists) is batched
     is_state_batched = bool(prep and prep.batch_size is not None)
     for op in circuit.operations[bool(prep) :]:
-        state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger)
-
+        if isinstance(op, Conditional):
+            if not mid_measurements[op.meas_val.measurements[0]]:
+                continue
+            op = op.then_op
```

Comment:
With the circuit `apply_operation` call signature, we could register a kernel for `Conditional` as well. But we can leave that as a TODO for now.

Reply:
I'll make the change.